detect/app.go

```diff
@@ -60,6 +63,16 @@ func GetAppNameFromGitRemote(directory string, remoteName string) (string, error
 	return "", errgo.Newf("[detect] Scalingo Git remote hasn't been found")
 }
 
+func extractAppNameFromCommandLine(c *cli.Context) string {
+	for _, context := range c.Lineage() {
```

Comment:
I'm not sure I understand what does `Lineage` do (https://pkg.go.dev/github.com/urfave/cli/v2#Context.Lineage), but I trust you if you tested it and it fixes the problem ^^

Shouldn't we also use it for the other global flags such as the region? I think it has the same problem, don't you think?

Reply:
There are two "app" flags defined: one global option, and one local to the actual command, both with default value "<name>".

`c.String("name")` in the context of a command returns the value for the local flag. When it is not set in the command line, it is the default value.

`c.Lineage()` returns an array with the local context, and all the parent contexts before, thus the root one with the global options.

So we can loop through that array to look for an `app` flag which has actually been set

Regarding the other `global` flags, indeed we have to investigate. I think the `addon` may be imacted as it uses a similar approach as the `app` one. For the other ones, more investigation are required